src/Database/DBStructure.php

```diff
@@ -664,12 +664,12 @@ public static function definition() {
 		$database["addon"] = [
 				"comment" => "registered addons",
 				"fields" => [
-						"id" => ["type" => "int", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => ""],
-						"name" => ["type" => "varchar(190)", "not null" => "1", "default" => "", "comment" => ""],
-						"version" => ["type" => "varchar(255)", "not null" => "1", "default" => "", "comment" => ""],
+						"id" => ["type" => "int unsigned", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => ""],
+						"name" => ["type" => "varchar(50)", "not null" => "1", "default" => "", "comment" => ""],
```

Comment:
I don't see a reason to always define the maximum length if we really don't need it. Of course there are tables that gain more than others from this change. But I want to apply the rules to *all* tables. I don't like this "oh, this table isn't *that* large, so we do not need to be that picky".

Reply:
I don't like rules for the sake of rules. I like rules to be meaningful, because rules are expensive to enforce. This is true for my kid, and this is true for coding as well. What if we actually need it someday?

Reply:
For other tables it really does make sense to restrict the lengths of the fields to only the needed size. This should help making the indexes smaller and therefore the queries faster. So the question is only if we should treat all tables the same way? And I think that we should. This makes it easier to implement if we don't have to think about "are we using this field in an index?" or "is this table queried a lot" and so on.

The size limits that I defined are having much room. So there shouldn't be any problems.

Reply:
Ok, let's move forward with this PR then. Do you intend on doing anything with Vagrant testing?

Reply:
I would like to move this task to the time of the release. Possibly we will have some other database structure changes until then as well.